File: app/src/main/java/ir/arapp/arappmain/Util/Adapters/NotificationAdapter.java
package ir.arapp.arappmain.Util.Adapters;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.makeramen.roundedimageview.RoundedImageView;
import com.ramotion.foldingcell.FoldingCell;

import java.util.ArrayList;

import ir.arapp.arappmain.Model.Notification;
import ir.arapp.arappmain.R;

public class NotificationAdapter extends RecyclerView.Adapter<NotificationAdapter.NotificationViewHolder>
{

//    region Variables
    ArrayList<Notification> notificationArrayList = new ArrayList<>();
//    endregion

//    region Methods
    @NonNull
    @Override
    public NotificationViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType)
    {
//        Inflate layout
        View itemView = LayoutInflater.from(parent.getContext()).inflate(R.layout.custom_notification_container, parent, false);
//        Return view
        return new NotificationViewHolder(itemView);
    }
    @Override
    public void onBindViewHolder(@NonNull NotificationViewHolder holder, int position)
    {
//        get current item in notification
        Notification currentNotification = notificationArrayList.get(position);
        if (notificationArrayList.size() == 0)
        {
            return;
        }
//        Set data and update xml
        holder.image.setImageResource(currentNotification.getImage());
        holder.title.setText(currentNotification.getTitle());
        holder.subtitle.setText(currentNotification.getSubtitle());
        holder.imageContext.setImageResource(currentNotification.getImageContext());
        holder.titleContext.setText(currentNotification.getTitle());
        holder.date.setText(currentNotification.getCreatedAt());
        holder.text.setText(currentNotification.getText());
        holder.foldingCell.setOnClickListener(view -> holder.foldingCell.toggle(false));
    }
    @Override
    public int getItemCount()
    {
        return notificationArrayList.size();
    }
//    Notify data changed
    public void setNotificationArrayList(ArrayList<Notification> notificationArrayList)
    {
        this.notificationArrayList = notificationArrayList;
        notifyDataSetChanged();
    }
//    endregion

//    View holder Class
    static class NotificationViewHolder extends RecyclerView.ViewHolder
    {
//        region Variables
        FoldingCell foldingCell;
        RoundedImageView image;
        RoundedImageView imageContext;
        TextView title;
        TextView titleContext;
        TextView subtitle;
        TextView text;
        TextView date;
//        endregion
//        Constructor to view holder
        public NotificationViewHolder(@NonNull View itemView)
        {
            super(itemView);
//            Hooks
            foldingCell = itemView.findViewById(R.id.foldingCell);
            image = itemView.findViewById(R.id.notificationImage);
            imageContext = itemView.findViewById(R.id.notificationImageContext);
            title = itemView.findViewById(R.id.notificationTitle);
            titleContext = itemView.findViewById(R.id.notificationTitleContext);
            subtitle = itemView.findViewById(R.id.notificationSubtitle);
            text = itemView.findViewById(R.id.notificationText);
            date = itemView.findViewById(R.id.notificationDate);
        }
    }
}

File: app/src/main/java/ir/arapp/arappmain/util/adapters/NotificationAdapter.kt
package ir.arapp.arappmain.util.adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.makeramen.roundedimageview.RoundedImageView
import com.ramotion.foldingcell.FoldingCell
import ir.arapp.arappmain.R
import ir.arapp.arappmain.model.base.Notification
import ir.arapp.arappmain.util.adapters.NotificationAdapter.NotificationViewHolder
import java.util.*

class NotificationAdapter : RecyclerView.Adapter<NotificationViewHolder>() {
    //    region Variables
    var notificationArrayList = ArrayList<Notification>()
        set(value) {
            field = value
            notifyDataSetChanged()
        }

    //    endregion
    //    region Methods
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): NotificationViewHolder {
//        Inflate layout
        val itemView = LayoutInflater.from(parent.context)
            .inflate(R.layout.custom_notification_container, parent, false)
        //        Return view
        return NotificationViewHolder(itemView)
    }

    override fun onBindViewHolder(holder: NotificationViewHolder, position: Int) {
//        get current item in notification
        val currentNotification = notificationArrayList[position]
        if (notificationArrayList.size == 0) {
            return
        }
        //        Set data and update xml
        holder.image.setImageResource(currentNotification.image)
        holder.title.text = currentNotification.title
        holder.subtitle.text = currentNotification.subtitle
        holder.imageContext.setImageResource(currentNotification.imageContext)
        holder.titleContext.text = currentNotification.title
        holder.date.text = currentNotification.createdAt
        holder.text.text = currentNotification.text
        holder.foldingCell.setOnClickListener { view: View? -> holder.foldingCell.toggle(false) }
    }

    override fun getItemCount(): Int {
        return notificationArrayList.size
    }

    //    Notify data changed


    //    endregion
    //    View holder Class
    inner class NotificationViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        //        region Variables
        var foldingCell: FoldingCell
        var image: RoundedImageView
        var imageContext: RoundedImageView
        var title: TextView
        var titleContext: TextView
        var subtitle: TextView
        var text: TextView
        var date: TextView

        //        endregion
        //        Constructor to view holder
        init {
            //            Hooks
            foldingCell = itemView.findViewById(R.id.foldingCell)
            image = itemView.findViewById(R.id.notificationImage)
            imageContext = itemView.findViewById(R.id.notificationImageContext)
            title = itemView.findViewById(R.id.notificationTitle)
            titleContext = itemView.findViewById(R.id.notificationTitleContext)
            subtitle = itemView.findViewById(R.id.notificationSubtitle)
            text = itemView.findViewById(R.id.notificationText)
            date = itemView.findViewById(R.id.notificationDate)
        }
    }
}